Configuration/Applications/python/cmsDriverOptions.py

```diff
@@ -244,13 +244,19 @@ def OptionsFromItems(items):
             raise Exception("Not a valid profiler type %s. Alternatives are pp, mp, fp=<function>."%(profilerType))
 
         options.prefix = "igprof -t cmsRun -%s" % profilerType
-        
+
     if options.heap_profile:
-        if options.profile and options.prefix:
+        if options.heap_profile and options.prefix:
             raise Exception("--heap_profile and --prefix are incompatible")
         profilerType = 'pp'
         options.prefix = "MALLOC_CONF=prof:true,prof_accum:true,prof_prefix:jeprof.out cmsRunJE "
 
+    if options.maxmem_profile:
+        if options.maxmem_profile and and options.prefix:
```

Comment:
Repetition of `maxmem_profile` seems unnecessary (it is already `True`; there is also second `and`)
```suggestion
        if options.prefix:
```